kotlin-checks-test-sources/src/main/kotlin/checks/UnsuitedFindFunctionWithNullComparisonCheckSample.kt

```diff
@@ -0,0 +1,95 @@
+package checks
+
+class UnsuitedFindFunctionWithNullComparisonCheckSample {
+
+    fun onCollections(list: List<Int>, set: Set<Int>, array: Array<Int>, map: Map<Int, Int>) {
+
+        list.find { it > 5 } != null // Noncompliant {{Replace "list.find { it > 5 } != null" with "list.any { it > 5 }".}}
+    //  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+        list.findLast { it > 5 } == null // Noncompliant  {{Replace "list.findLast { it > 5 } == null" with "list.none { it > 5 }".}}
+
+        list.find { it == 5 } != null // Noncompliant  {{Replace "list.find { it == 5 } != null" with "list.contains(5)".}}
+        list.find { 5 == it } != null // Noncompliant  {{Replace "list.find { 5 == it } != null" with "list.contains(5)".}}
+        list.find { it != 5 } == null // Noncompliant  {{Replace "list.find { it != 5 } == null" with "list.contains(5)".}}
+        list.find { 5 != it } == null // Noncompliant  {{Replace "list.find { 5 != it } == null" with "list.contains(5)".}}
+
+        list.find { it == 5 } == null // Noncompliant  {{Replace "list.find { it == 5 } == null" with "!list.contains(5)".}}
+        list.find { 5 == it } == null // Noncompliant  {{Replace "list.find { 5 == it } == null" with "!list.contains(5)".}}
+        list.find { it != 5 } != null // Noncompliant  {{Replace "list.find { it != 5 } != null" with "!list.contains(5)".}}
```

Comment:
I don't think this is quite correct. `list.find { it != 5 } != null` is checking if the list contains an element other than 5. It doesn't matter if 5 is in it, si I think `list.any { it != 5 }` would be a more accurate replacement.

Reply:
You're right this will diverge in case of an empty collection:
`!list.contains(5)` returns true, while `list.find { it != 5 } != null` return false! Nice catch! :)